test/goalc/test_with_game.cpp

```diff
@@ -0,0 +1,116 @@
+#include <thread>
+#include <chrono>
+
+#include "gtest/gtest.h"
+#include "game/runtime.h"
+#include "goalc/listener/Listener.h"
+#include "goalc/compiler/Compiler.h"
+
+#include "third-party/inja.hpp"
+#include "third-party/json.hpp"
+#include <common\util\FileUtil.h>
+
+#include <test\goalc\framework\test_runner.h>
+
+#include <iostream>
+#include <string>
+#include <cstdio>
+#include <sstream>
+#include <iostream>
+#include <random>
+#include <filesystem>
+
+struct WithGameParam {
+  // TODO - Not Needed Yet
+};
+
+class WithGameTests : public testing::TestWithParam<WithGameParam> {
+ public:
+  static void SetUpTestSuite() {
+    runtime_thread = std::thread((GoalTest::runtime_no_kernel));
+    runner.c = &compiler;
+  }
+
+  static void TearDownTestSuite() {
+    compiler.shutdown_target();
+    runtime_thread.join();
+  }
+
+  void SetUp() {
+    GoalTest::createDirIfAbsent(GoalTest::getTemplateDir(testCategory));
+    GoalTest::createDirIfAbsent(GoalTest::getGeneratedDir(testCategory));
+  }
+
+  void TearDown() {}
+
+  static std::thread runtime_thread;
+  static Compiler compiler;
+  static GoalTest::CompilerTestRunner runner;
+
+  std::string testCategory = "with_game";
+  inja::Environment env{GoalTest::getTemplateDir(testCategory),
+                        GoalTest::getGeneratedDir(testCategory)};
+};
+
+std::thread WithGameTests::runtime_thread;
+Compiler WithGameTests::compiler;
+GoalTest::CompilerTestRunner WithGameTests::runner;
+
+// TODO - havn't done anything with these really yet
+TEST_F(WithGameTests, All) {
```

Comment:
This is probably for another PR, but I think a lot of these tests just call a single function with a few arguments and check the result. I imagine we could have a Call Function test that generates these without needing all these separate test `.gc` files.